packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,29 +190,297 @@ class CupertinoAlertDialog extends StatelessWidget {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+
+    final double contentWidth = _content.getMaxIntrinsicWidth(height);
+    final double actionsWidth = _actions.getMaxIntrinsicWidth(height);
+    final double width = min(contentWidth, actionsWidth);
+
+    if (width.isFinite)
+      return width;
+    return 0.0;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      CupertinoDialogRenderBoxParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    final double contentHeight = _content.getMaxIntrinsicHeight(width);
+    final double actionsHeight = _actions.getMaxIntrinsicHeight(width);
+    final double height = max(contentHeight, actionsHeight);
+
+    if (height.isFinite)
+      return height;
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      CupertinoDialogRenderBoxParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    final double contentHeight = _content.getMaxIntrinsicHeight(width);
+    final double actionsHeight = _actions.getMaxIntrinsicHeight(width);
+    final double height = min(contentHeight, actionsHeight);
+
+    if (height.isFinite)
+      return height;
+    return 0.0;
+  }
+
+  @override
+  void performLayout() {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      CupertinoDialogRenderBoxParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    // TODO(mattcarroll): parameterize actions count.
+    final actionsCount = 3;
+
+    double minActionSpace = 0.0;
+    if (actionsCount > 0 && actionsCount <= 2) {
+      minActionSpace = _kButtonHeight;
+    } else {
+      minActionSpace = 1.5 * _kButtonHeight;
+    }
+
+    final size = constraints.constrain(const Size(double.infinity, double.infinity));
+
+    // Size alert dialog content.
+    final Size maxContentSize = new Size(size.width, size.height - minActionSpace);
+    _content.layout(
+      new BoxConstraints.loose(maxContentSize),
```

Comment:
ah, you might not need to convert the constraints to a size and back. 

You can probably just do constraints.deflate(new EdgeInsets(top: minActionSpace))

Reply:
Looks like we probably can do that.  But should we?  Does that code better express the intention of the calculation than what is there now?